### PR TITLE
Fix rr_graph direct link

### PR DESCRIPTION
**Pull request by amin1377**
This PR is created to fix issue #2145
@vaughnbetz

### CODE REVIEW

File: vpr/src/route/rr_graph.cpp [hunk 1]
```diff
@@ -2962,12 +2962,20 @@ static int get_opin_direct_connections(RRGraphBuilder& rr_graph_builder,
                             }
                         }
 
-                        int target_sub_tile = z + directs[i].sub_tile_offset;
-                        if (relative_ipin >= target_type->sub_tiles[target_sub_tile].num_phy_pins) continue;
+                        int target_cap = z + directs[i].sub_tile_offset;
```
Comment (vaughnbetz) [inline vpr/src/route/rr_graph.cpp:2965]: Can you comment what the logic is here?

File: vpr/src/route/rr_graph.cpp [hunk 2]
```diff
@@ -2962,12 +2962,20 @@ static int get_opin_direct_connections(RRGraphBuilder& rr_graph_builder,
...
+                                break;
+                            }
+                        }
+                        VTR_ASSERT(target_sub_tile != nullptr);
+                        if (relative_ipin >= target_sub_tile->num_phy_pins) continue;
 
                         //If this block has capacity > 1 then the pins of z position > 0 are offset
                         //by the number of pins per capacity instance
-                        int ipin = get_physical_pin_from_capacity_location(target_type, relative_ipin, target_sub_tile);
+                        int ipin = get_physical_pin_from_capacity_location(target_type, relative_ipin, target_cap);
```
Comment (vaughnbetz) [inline vpr/src/route/rr_graph.cpp:2978]: Is the comment above still correct?  I'm not seeing the logic of how the pins wrap around (pins of z position > 0 offset by number of pins per capacity instance).

**Review by vaughnbetz — COMMENTED** [review]
Nice job finding this!  Can you update the comments in the code to explain the logic though and make sure the existing comment is still correct?

File: vpr/src/route/rr_graph.cpp (hunk 2 above)
Comment (amin1377) [reply vpr/src/route/rr_graph.cpp:2978]: "relative_pin" shows the pin number in its capacity. So, to get the number of the pin at tile-level, "relative_ipin" should be added by the number of pins on prior sub_tiles and capacities.

File: vpr/src/route/rr_graph.cpp (hunk 1 above)
Comment (amin1377) [reply vpr/src/route/rr_graph.cpp:2967]: Done

**Pull request merged**